src/devices/Sn74hc595/GpioControllerExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+
+/// <summary>
+/// Extension methods for GpioController
+/// </summary>
+public static class GpioControllerExtensions
+{
+    /// <summary>
+    /// Opens a set of pins.
+    /// </summary>
+    public static void OpenPins(this GpioController controller, PinMode mode, params int[] pins)
```

Comment:
I would be interested in your opinion of why do you think these extensions were required, to see if we have a gap in our API. Specifically abtout your OpenPinAndWrite, I wonder if our Open methods should have an optional arg Initial value

Reply:
Excellent q. I agree that that API option/question you suggest/put on the table is a better idea. I think we should do that or nothing at all. The only challenge is that the initial value is only relevant for one PinMode type.